src/services/getEditsForFileRename.ts

```diff
@@ -3,41 +3,35 @@ namespace ts {
-            for (const { sourceFile, toUpdate } of getImportsToUpdate(program, oldFilePath, host)) {
-                const newPath = pathUpdater(isRef(toUpdate) ? toUpdate.fileName : toUpdate.text);
-                if (newPath !== undefined) {
-                    const range = isRef(toUpdate) ? toUpdate : createStringRange(toUpdate, sourceFile);
-                    changeTracker.replaceRangeWithText(sourceFile, range, isRef(toUpdate) ? newPath : removeFileExtension(newPath));
-                }
-            }
+            updateTsconfigFiles(program, changeTracker, pathUpdater);
+            updateImports(program, changeTracker, pathUpdater, host);
         });
     }
 
-    function updateTsconfigFiles(program: Program, changeTracker: textChanges.ChangeTracker, oldFilePath: string, newFilePath: string): void {
+    function updateTsconfigFiles(program: Program, changeTracker: textChanges.ChangeTracker, pathUpdater: PathUpdater): void {
         const configFile = program.getCompilerOptions().configFile;
-        const oldFile = getTsConfigPropArrayElementValue(configFile, "files", oldFilePath);
-        if (oldFile) {
-            changeTracker.replaceRangeWithText(configFile, createStringRange(oldFile, configFile), newFilePath);
-        }
-    }
+        for (const property of getTsConfigPropArray(configFile, "files")) {
+            if (!isArrayLiteralExpression(property.initializer)) continue;
 
-    interface ToUpdate {
-        readonly sourceFile: SourceFile;
-        readonly toUpdate: StringLiteralLike | FileReference;
-    }
-    function isRef(toUpdate: StringLiteralLike | FileReference): toUpdate is FileReference {
-        return "fileName" in toUpdate;
+            for (const element of property.initializer.elements) {
+                if (!isStringLiteral(element)) continue;
+
+                const updated = pathUpdater(element.text, element.text, /*isImport*/ false);
+                if (updated !== undefined) {
+                    changeTracker.replaceRangeWithText(configFile, createStringRange(element, configFile), updated);
+                }
+            }
+        }
     }
 
-    function getImportsToUpdate(program: Program, oldFilePath: string, host: LanguageServiceHost): ReadonlyArray<ToUpdate> {
+    function updateImports(program: Program, changeTracker: textChanges.ChangeTracker, pathUpdater: PathUpdater, host: LanguageServiceHost): void {
         const checker = program.getTypeChecker();
-        const result: ToUpdate[] = [];
         for (const sourceFile of program.getSourceFiles()) {
             for (const ref of sourceFile.referencedFiles) {
-                if (!program.getSourceFileFromReference(sourceFile, ref) && resolveTripleslashReference(ref.fileName, sourceFile.fileName) === oldFilePath) {
-                    result.push({ sourceFile, toUpdate: ref });
-                }
+                if (program.getSourceFileFromReference(sourceFile, ref)) continue;

```

Comment:
why would you want to skip if ref was resolved to some sourceFile? 
if you had 
index.ts:
```typescript
/// <ref path="file1.ts"/>
```

file1.ts
```typescript
// Some content
```
You can rename file1.ts to file2.ts and want to change ref in index to file2.ts?

Reply:
You're right, I did a similar thing in #24328 but I'll have to update it in this PR too.